src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.h

```diff
@@ -1386,22 +1390,23 @@ ep_rt_utf8_to_utf16le_string (
     if (!str)
         return NULL;
 
-    // Shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-    // Implementation would just use strlen and malloc to make a new buffer, and would then copy the string chars one by one.
-    // Assumes that only ASCII is used for ep_char8_t
-    size_t len_utf8 = strlen(str);
-    ep_char16_t *str_utf16 = reinterpret_cast<ep_char16_t *>(malloc ((len_utf8 + 1) * sizeof (ep_char16_t)));
-    if (!str_utf16)
+    int32_t flags = MINIPAL_MB_NO_REPLACE_INVALID_CHARS;
+
+    ep_char16_t* lpDestStr = NULL;
+
+    if (static_cast<int>(len) < 0)
```

Comment:
Does eventpipe actually use this "feature" where negative length means to compute the length? It would be best to avoid it unless it is required.

Reply:
We can remove these runtime-specific wrappers of `ep_rt_utf8_to_utf16le_string` and move it to src/native/eventpipe; now that both [coreclr (via win-PAL)](https://github.com/dotnet/runtime/blob/2811e7c7551c5ec56f1afbfd1749e1f42d264dbf/src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h#L1435) and [mono (via eglib)](https://github.com/dotnet/runtime/blob/2811e7c7551c5ec56f1afbfd1749e1f42d264dbf/src/mono/mono/eventpipe/ep-rt-mono.h#L1385) end up calling the same minipal APIs which this implementation directly invokes.

Reply:
There are patterns in EventPipe library where -1 is used to compute the length, such as [here](https://github.com/dotnet/runtime/blob/main/src/native/eventpipe/ep-event-source.c#L111).

Reply:
Ok, I would change these checks to `if (len == (size_t)-1)` so that they match `-1` only.